### PR TITLE
feat: rename logrlint to loggercheck

### PR DESCRIPTION
**Pull request by timonwong**
Since `logrlint` reaches v0.1.0, more loggers are supported (`klog`, `zap`). 

So as suggested in #3140, `logrlint` is now deprecated, but since it's just a repo rename, so logrlint is renamed to [loggercheck](https://github.com/timonwong/loggercheck) using alternate names.

New features:
- checks for klog and zap
- allow custom rules
- allow disabling checks individually
- allow check for logging key (require string literal/constant, ascii string)
- allow check for printf-like specifiers (%s, %d, etc)

### CODE REVIEW

**Comment by ldez** [comment]
you cannot rename a linter.

https://github.com/golangci/golangci-lint/pull/3140#issuecomment-1228343049

**Comment by ldez** [comment]
I will apply some changes to your PR.

File: .golangci.reference.yml [hunk 1]
```diff
@@ -1111,6 +1111,11 @@ linters-settings:
     # Default: 1
     tab-width: 1
 
+  loggercheck:
+    logr: true
+    klog: true
+    zap: true
+
```
Comment (ldez) [inline .golangci.reference.yml:1118]: If the linter has a configuration, the exhaustive configuration of the linter must be added (alphabetical case-insensitive order)
    - The values must be different from the default ones.
    - The default values must be defined in a comment.
    - The option must have a short description.

**Comment by ldez** [comment]
Please wait, I haven't seen that you have used the alternative names.

So you can "rename".

**Comment by ldez** [comment]
@timonwong you can do your rename, read my previous message

**Comment by timonwong** [comment]
> Please wait, I haven't seen that you have used the alternative names.
> 
> So you can "rename".

@ldez Ok, I'll do it right now

**Review by ldez — CHANGES_REQUESTED** [review]

**Comment by ldez** [comment]
just a quick note about klog and kubernetes https://github.com/kubernetes-sigs/logtools/tree/main/logcheck

File: .golangci.reference.yml [hunk 2]
```diff
@@ -1111,6 +1111,22 @@ linters-settings:
...
+    # Default: true
+    klog: false
+    # Allow check for the github.com/go-logr/logr library.
+    # Default: true
+    logr: false
+    # Allow check for the "sugar logger" from go.uber.org/zap library.
+    # Default: false
+    zap: true
+    # List of rules to check against, where each rule is a single logger pattern.
+    rules:
```
Comment (ldez) [inline .golangci.reference.yml:1125]: what is the default value? Please add the value to the file.

**Review by ldez — CHANGES_REQUESTED** [review]

File: .golangci.reference.yml (hunk 2 above)
Comment (timonwong) [reply .golangci.reference.yml:1126]: By default the (custom) rules is empty

**Comment by timonwong** [comment]
> just a quick note about klog and kubernetes https://github.com/kubernetes-sigs/logtools/tree/main/logcheck

Hmmm, if knowing this before i may not create another linter myself :joy:

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**